libcaf_net/caf/net/http/request_header.cpp

```diff
@@ -10,28 +13,16 @@ namespace {
 
 constexpr std::string_view eol = "\r\n";
 
-template <class F>
-bool for_each_line(std::string_view input, F&& f) {
-  auto input_end = input.data() + input.size();
-  for (auto pos = input.data();;) {
-    auto line_end = std::search(pos, input_end, eol.begin(), eol.end());
-    if (line_end == input_end || std::distance(pos, input_end) == 2) {
-      // Reaching the end or hitting the last empty line tells us we're done.
-      return true;
-    }
-    auto to_line_end = std::distance(pos, line_end);
-    CAF_ASSERT(to_line_end >= 0);
-    if (!f(std::string_view{pos, static_cast<size_t>(to_line_end)}))
-      return false;
-    pos = line_end + eol.size();
-  }
-  return true;
-}
-
 } // namespace
 
-request_header::request_header(const request_header& other) {
-  assign(other);
+request_header::request_header(const request_header& other) : header{other} {
+  if (other.valid()) {
+    method_ = other.method_;
+    uri_ = other.uri_;
+    version_ = remap(other.raw_.data(), other.version_, raw_.data());
+  } else {
+    version_ = std::string_view{};
+  }
```

Comment:
That's already the default behavior. No need to spell it out.

```suggestion
  }
```